common/unified/solver/pipe_cg_kernels.cpp

```diff
@@ -119,21 +117,21 @@ void step_1(std::shared_ptr<const DefaultExecutor> exec,
     // w = w - tmp * g
     run_kernel_solver(
         exec,
-        [] GKO_KERNEL(auto row, auto col, auto x, auto r, auto z, auto w,
-                      auto p, auto q, auto f, auto g, auto rho, auto beta,
-                      auto stop) {
+        [] GKO_KERNEL(auto row, auto col, auto x, auto r, auto z1, auto z2,
+                      auto w, auto p, auto q, auto f, auto g, auto rho,
+                      auto beta, auto stop) {
             if (!stop[col].has_stopped()) {
                 auto tmp = safe_divide(rho[col], beta[col]);
                 x(row, col) += tmp * p(row, col);
                 r(row, col) -= tmp * q(row, col);
-                z(row, col) -= tmp * f(row, col);
+                z1(row, col) -= tmp * f(row, col);
+                z2(row, col) -= tmp * f(row, col);
```

Comment:
```suggestion
                z1(row, col) -= tmp * f(row, col);
                z2(row, col) = z1(row, col);
```
Because it is always synced, so using assign should be better

Reply:
yes, so it really does cause some trouble if we use an assignment operator here

Reply:
Could you elaborate the trouble?